### PR TITLE
[Coin Modularization] Tezos

### PR DESCRIPTION
**Pull request by sprohaszka-ledger**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

Tezos in its own package.

The code has also been reorganise to move forward like Polkadot.

Some issue are still to discuss:
* Tezos `Ledger-signer` package is used in `prepareTransaction`
    * Therefore to avoid import `ledgerhq/hw-transport` an `overrides` directive has been added to `package.json` file
* `encodeAddress` has "FIXME" to see as previous code didn't pass Typescript strict checking

~~As a side effect to this PR, `SignerContext` has been simplified (to be more like the dedicated Bitcoin definition). Remove of the return composition type `CoinAddress` and `CoinSignature` as a mandatory declaration.
Thus other coin-modules contains this modification (no impact in their behaviour)~~ *This feature has been separated to a different [PR](https://github.com/LedgerHQ/ledger-live/pull/6844)*

### ❓ Context

- **JIRA or GitHub link**: [LIVE-8177]

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

[LIVE-8177]: https://ledgerhq.atlassian.net/browse/LIVE-8177?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/coin-modules/coin-evm/src/bridge/js.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import {
 import { SignerContext } from "@ledgerhq/coin-framework/signer";
 import type { AccountBridge, Bridge, CurrencyBridge } from "@ledgerhq/types-live";
 import getAddressWrapper from "@ledgerhq/coin-framework/bridge/getAddressWrapper";
-import type { EvmAddress, EvmSignature, EvmSigner } from "../types/signer";
```
Comment (gre) [inline libs/coin-modules/coin-evm/src/bridge/js.ts:10]: shouldn't modification of coin-evm/*  be done in a separated PR? it's no longer a PR about tezos 🤔

**Review by gre — DISMISSED** [review]
ok for hub scope

File: libs/coin-modules/coin-evm/src/bridge/js.ts (hunk 1 above)
Comment (sprohaszka-ledger) [reply libs/coin-modules/coin-evm/src/bridge/js.ts:10]: I wrote a general comment about this: "As a side effect to this PR, SignerContext has been simplified (to be more like the dedicated Bitcoin definition)".
I might separate it though...

**Review by hedi-edelbloute — DISMISSED** [review]

File: libs/coin-modules/coin-tezos/src/types/signer.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+export type TezosSignature = {
+  signature: string;
+};
+// Type coming from hw-app-tezos
+export const TezosCurves = {
+  ED25519: 0x00,
+  SECP256K1: 0x01,
+  SECP256R1: 0x02,
+};
+export type Curve = (typeof TezosCurves)[keyof typeof TezosCurves];
```
Comment (lvndry) [inline libs/coin-modules/coin-tezos/src/types/signer.ts:14]:
```suggestion
type TezosCurves = {
  ED25519: 0x00;
  SECP256K1: 0x01;
  SECP256R1: 0x02;
};

export type Curve = TezosCurves[keyof TezosCurves];
```

TezoCurves is not used outside of module no need to export it

File: libs/coin-modules/coin-tezos/src/types/signer.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    rawTxHex: string,
+    options: {
+      curve?: Curve;
+    },
+  ): Promise<TezosSignature>;
+  // Tezos [LedgerSigner](https://www.npmjs.com/package/@taquito/ledger-signer)
+  createLedgerSigner(path: string, prompt: boolean, derivationType: number): LedgerSigner;
+}
+
+// export type SignerContext = <T>(
```
Comment (lvndry) [inline libs/coin-modules/coin-tezos/src/types/signer.ts:50]: Can be deleted

File: libs/coin-framework/src/signer.ts [hunk 4]
```diff
@@ -2,4 +2,4 @@
 // The transport is passed to the method provided and then closed.
 // So we can't keep an instance depending on the transport, as it will usefull outside of the `job`.
 export type PassthroughFn<T, U> = (signer: T) => Promise<U>;
-export type SignerContext<T, U> = (deviceId: string, fn: PassthroughFn<T, U>) => Promise<U>;
+export type SignerContext<T> = <U>(deviceId: string, fn: PassthroughFn<T, U>) => Promise<U>;
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply libs/coin-framework/src/signer.ts:5]: If you take a look at all the `SignerContext` usage you will notice that `U` is no longer required to be defined and can be infered by the compiler, regarding the output type of `fn`.

File: libs/coin-modules/coin-algorand/src/hw-getAddress.ts [hunk 5]
```diff
@@ -1,15 +1,11 @@
...
 
-const resolver = (
-  signerContext: SignerContext<AlgorandSigner, AlgorandAddress | AlgorandSignature>,
-): GetAddressFn => {
+const resolver = (signerContext: SignerContext<AlgorandSigner>): GetAddressFn => {
   return async (deviceId: string, { path, verify }: GetAddressOptions) => {
-    const r = (await signerContext(deviceId, signer =>
-      signer.getAddress(path, verify || false),
-    )) as AlgorandAddress;
+    const r = await signerContext(deviceId, signer => signer.getAddress(path, verify || false));
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply libs/coin-modules/coin-algorand/src/hw-getAddress.ts:8]: Nope. Unnecessary, we don't want this value to be resolved right now (in this part of the code).

File: libs/coin-modules/coin-tezos/src/bridge/transactionStatus.ts [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+  RecommendUndelegation,
+} from "@ledgerhq/errors";
+import { log } from "@ledgerhq/logs";
+import { TezosAccount, Transaction, TransactionStatus } from "../types";
+import { InvalidAddressBecauseAlreadyDelegated } from "../types/errors";
+import api from "../api/tzkt";
+import { isAccountDelegating } from "../api/bakers";
+import { validateRecipient } from "./prepareTransaction";
+
+const EXISTENTIAL_DEPOSIT = new BigNumber(275000);
```
[the remark this answers is not in the document]
Comment (sprohaszka-ledger) [reply libs/coin-modules/coin-tezos/src/bridge/transactionStatus.ts:18]: Nope, I am not a Tezos connoisseur 🤷‍♂️

**Review by lvndry — APPROVED** [review]

**Review by lambertkevin — APPROVED** [review]
🙈

**Pull request merged**